Add docs for the configuration input detection opt-out flags for Gradle 8.3 (#25757)
gradle/gradle: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

Having new configuration misses because of them may be undesirable for the build users, and the suggested strategy for eliminating them is:

* Identify the configuration inputs causing the invalidation of the configuration cache with the help of the <<configuration_cache#config_cache:troubleshooting, configuration cache report>>.
** Fix undeclared configuration inputs accessed by the build logic of the project.
** Report issues caused by third-party plugins to the plugin maintainers, and update the plugins once they get fixed.

* For some kinds of configuration inputs, it is possible to use the opt-out options that make Gradle fall back to the earlier behavior, omitting the inputs from detection.
This temporary workaround is aimed to mitigate performance issues coming from out-of-date plugins.

It is currently possible to temporarily opt out of configuration input detection in the following case:

* Since Gradle 8.1, using many APIs related to the file system is correctly tracked as configuration inputs, including the file system checks, such as `File.exists()` or `File.isFile()`.
+

> For the input tracking to ignore these file system checks on the specific paths, one can provide a Gradle property `org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks` with the list of the paths, relative to the root project directory and separated by `;`.

Reviewer: ```suggestion
For the input tracking to ignore these file system checks on the specific paths, the Gradle property `org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks`, with the list of the paths, relative to the root project directory and separated by `;`, can be provided.
```

To ignore multiple paths, use `*` to match arbitrary strings within one segment, or `pass:[**]` across segments.

> Paths starting with `~/` will be considered to be based on the user home directory.

Reviewer: ```suggestion
Paths starting with `~/` are based on the user home directory.
```

For example:
+
[source,properties]
.gradle.properties
----
org.gradle.configuration-cache.inputs.unsafe.ignore.file-system-checks=\
    ~/.third-party-plugin/*.lock;\
    ../../externalOutputDirectory/**;\
    build/analytics.json
----

> One should ignore configuration inputs sparingly, and only if those do not affect the tasks produced by the configuration logic.

Reviewer: ```suggestion
Ignore configuration inputs sparingly, and only if they do not affect the tasks produced by the configuration logic.
```
